### PR TITLE
Device id and signup database linking

### PR DESCRIPTION
**Pull request by ichandel**

### CODE REVIEW

**Review by DanTang212 — APPROVED** [review]
Looks good

File: src/app/src/main/java/com/example/myapplication/WelcomeActivity.java [hunk 1]
```diff
@@ -28,20 +36,46 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-                Intent i = new Intent(WelcomeActivity.this, SignUpActivity.class);
-                startActivity(i);
-                finish();
+        String deviceId = Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID);
 
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
 
-            }
+        db.collection("users").document(deviceId).get()
```
Comment (Cedric-Boucher) [inline src/app/src/main/java/com/example/myapplication/WelcomeActivity.java:47]: please use the DatabaseManager instead as it takes care of everything for you and builds everything connected to the user

**Review by Cedric-Boucher — CHANGES_REQUESTED** [review]

**Review by Cedric-Boucher — COMMENTED** [review]
the rest looks alright I think

File: src/app/src/main/java/com/example/myapplication/WelcomeActivity.java (hunk 1 above)
Comment (ichandel) [reply src/app/src/main/java/com/example/myapplication/WelcomeActivity.java:47]: i'll update it to reflect your requested changes, it was just redundant as we didn't need to build a user object here, just merely flag whether it was present in the database or not

**Review by alianish11 — APPROVED** [review]

File: src/app/src/main/java/com/example/myapplication/WelcomeActivity.java [hunk 2]
```diff
@@ -28,20 +37,70 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
+        String deviceId = Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID);
 
-            }
+        dbManager.getUser(deviceId, this);
 
-        });
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+        db.collection("users").document(deviceId).get()
```
Comment (Cedric-Boucher) [inline src/app/src/main/java/com/example/myapplication/WelcomeActivity.java:52]: this code is still here even though the DatabaseManager stuff has now been added?

**Review by Cedric-Boucher — CHANGES_REQUESTED** [review]

File: src/app/src/main/java/com/example/myapplication/WelcomeActivity.java (hunk 2 above)
Comment (Cedric-Boucher) [reply src/app/src/main/java/com/example/myapplication/WelcomeActivity.java:52]: I'm just confused why there's both now?

Comment (ichandel) [reply src/app/src/main/java/com/example/myapplication/WelcomeActivity.java:52]: wait thats weird, i got rid of it
lemme try pushing again

**Review by Cedric-Boucher — APPROVED** [review]

File: src/app/src/main/java/com/example/myapplication/WelcomeActivity.java (hunk 2 above)
Comment (ichandel) [reply src/app/src/main/java/com/example/myapplication/WelcomeActivity.java]: should be fixed now

**Review by alianish11 — APPROVED** [review]

**Pull request merged**